virt/virtual_machines/virt-accessing-vm-ssh.adoc

```diff
@@ -0,0 +1,84 @@
+:_content-type: ASSEMBLY
+[id="virt-accessing-vm-ssh"]
+= Using SSH to access virtual machines
+include::_attributes/common-attributes.adoc[]
+:context: virt-accessing-vm-ssh
+
+toc::[]
+
+You can configure an SSH connection to access virtual machines (VMs) by using the following methods:
+
+* `virtctl ssh` command
++
+You add a public SSH key to a VM before it boots for the first time. Then, you connect to the VM by using the `virtctl ssh` command.
++
+This method is useful for troubleshooting a VM because it requires little time to set up. The connection is slow, but it works consistently even if you are behind a firewall or if the cluster does not have a load balancer.
++
+[NOTE]
+====
+Only the user who created the SSH key pair can use the `virtctl ssh` command.
+====
+
+// dynamic key injection - CNV-30068
+// OpenSSH and virtctl - CNV-26737
+
+* Kubernetes service
++
+You create a service to expose a VM port, add the service to the VM, and then connect by using the `ssh` command.
++
+This method is more time-consuming to set up but it provides better performance than `virtctl`.
+
+[id="connecting-ssh-virtctl_virt-accessing-vm-ssh"]
+== Using virtctl for SSH connections
+
+You create an SSH key pair and add the public key to a virtual machine (VM) before it boots for the first time. Then, you run the `virtctl ssh` command.
+
+Only one user can use the `virtctl ssh` command because the command points to the private key of the SSH key pair.
+
+[id="adding-public-keys-vm_virt-accessing-vm-ssh"]
+=== Adding public keys to virtual machines
+
+You add a _static_ public key by using the {product-title} web console or the command line.
+
+The key must be added before the virtual machine (VM) boots for the first time.
+
+include::modules/virt-adding-static-public-key-web.adoc[leveloffset=+3]
+
+[id="adding-static-public-key-cli_virt-accessing-vm-ssh"]
+==== Adding public keys to virtual machines from the command line
+
+Add a _static_ public key from the command line before the virtual machine (VM) boots for the first time.
+
+include::modules/virt-creating-ssh-secret.adoc[leveloffset=+4]
+
+include::modules/virt-adding-static-public-key-cli.adoc[leveloffset=+4]
+
+include::modules/virt-running-virtctl-ssh-command.adoc[leveloffset=+2]
+
+.Copying the `virtctl ssh` command
+
+You copy the `virtctl ssh` command in the web console by selecting *Copy SSH command* from the Options menu beside a VM on the xref:../../virt/getting_started/virt-web-console-overview.adoc#ui-virtualmachines-page[*VirtualMachines* page].
+
+[id="connecting-ssh-services_virt-accessing-vm-ssh"]
+== Using services for SSH connections
+
+You configure a service to expose a port and then connect by using the `SSH` command to a virtual machine (VM). Services provide the best performance and are recommended for high-traffic applications.
+
```

Comment:
```
Using services for SSH connections

You configure a service to expose a port and then connect by using the SSH command to a virtual machine (VM). Services provide the best performance and are recommended for high-traffic applications.
````

@phoracek is this correct? I would have expected

```
Using services for SSH connections

You configure a service to expose a port and then connect by using the SSH command to a virtual machine (VM). Seconday networks utilizing SR-IOV provide the best performance and are recommended for high-traffic applications.
```

Reply:
IMO regular Services (with type LoadBalancer) are more likely to be found in a regular cluster. They provide good performance. I would not expect SR-IOV to make a huge difference for SSH connections.

Reply:
@dominikholler re: "single-node bottlenecking" - are you referring to SNO?